lib/ansible/module_utils/network/iosxr/iosxr.py

```diff
@@ -454,7 +454,10 @@ def run_command(module, commands):
             sendonly = False
             newline = True
 
-        out = conn.get(command, prompt=prompt, answer=answer, sendonly=sendonly, newline=newline)
+        try:
+            out = conn.get(command=command, prompt=prompt, answer=answer, sendonly=sendonly, newline=newline)
+        except ConnectionError as e:
```

Comment:
Why the ConnectionError exception is returned as a normal response here. In case of an error, this might not be right.

Reply:
if there is an exception in conn.get then out is not defined so it is going to fail below at responses.append(to_text(out, errors='surrogate_or_strict')) so with this fix at least we will get some info .btw ios's run_command have similar check

Reply:
I see your point too..returning it as normal response might not be good . better to raise the exception in case it has hit connectionError Exception.